src/_zkapauthorizer/recover.py

```diff
@@ -83,6 +83,9 @@ def marshal(self) -> dict[str, Optional[str]]:
 # An object which can retrieve remote ZKAPAuthorizer state.
 Downloader = Callable[[SetState], Awaitable[BinaryIO]]
 
+# function which can set remote ZKAPAuthorizer state.
+Uploader = Callable[[SetState], Awaitable[None]]
```

Comment:
Can you put `Uploader` and the related upload functionality below into the new `replicate.py`?